src/setup/node.rs

```diff
@@ -1,20 +1,30 @@
 use std::{
     fs, io,
-    net::{IpAddr, SocketAddr},
+    net::{IpAddr, SocketAddr, SocketAddrV4},
     path::PathBuf,
-    process::{Child, Command, Stdio},
-    time::Duration,
+    process::Child,
 };
 
 use anyhow::Result;
-use tokio::io::AsyncWriteExt;
+use fs_extra::dir::{copy, CopyOptions};
+use tempfile::TempDir;
 
 use crate::{
-    setup::config::{NodeConfig, NodeMetaData, RippledConfigFile, RIPPLED_CONFIG, RIPPLED_DIR},
-    wait_until,
+    setup::{
+        config::{
+            NewNodeConfig, NodeMetaData, RippledConfigFile, RIPPLED_CONFIG, RIPPLED_DIR,
+            ZIGGURAT_DIR,
+        },
+        process,
+        process::wait_for_start,
+    },
+    tools::constants::{JSON_RPC_PORT, STATEFUL_DIR, STATEFUL_IP},
 };
 
-pub const CONNECTION_TIMEOUT: Duration = Duration::from_secs(2);
+pub enum NodeConfig {
```

Comment:
ah, I see 👍

Reply:
How about calling this `NodeSetup` instead?

Reply:
Done